### PR TITLE
Change Netlify to deploy both UI and website

### PR DESCRIPTION
**Pull request by backspace**
This makes Netlify build both `ui/` and `website/`. I’ve extracted the
build steps to their own script, as it was getting unwieldy to have
it all jammed into a single line in `netlify.toml`.

The redirects are dynamic based on whether the branch being deployed
is determined to be for the UI. When that’s the case, visiting `/`
will automatically forward you to `/ui`, which was the behaviour before
this PR. For all other PRs, visiting `/` will show the documentation
root and you’ll have to manually append `/ui` to see the UI.

Netlify deployments with this scheme are taking significantly longer
as it appears that caching of dependencies isn’t supported for
subdirectories, so the gems and npm packages are being fetched and
rebuilt every time. Some background here:
https://github.com/netlify/build-image/issues/196

### CODE REVIEW

**Comment by backspace** [comment]
Re the conditional redirection, you can see it in action by visiting the [deployment for this branch](https://deploy-preview-6167--nomad-ui.netlify.com/) vs [the one](https://deploy-preview-6168--nomad-ui.netlify.com/) for [this branch](https://github.com/hashicorp/nomad/pull/6168), which I named to follow the UI pattern. The latter jumps directly to `/ui`.

**Comment by backspace** [comment]
I’d return this to draft if I could; deployments are now timing out due to exceeding a 15-minute limit. I’ve written to Netlify support.

File: .netlify/build-and-move.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+
+# Build the static web site in website/build
+# bundle install     should be cached???
```
Comment (DingoEatingFuzz) [inline .netlify/build-and-move.sh]: Looks like you got to the bottom of this in finding https://github.com/netlify/build-image/issues/196

Can you update the comment to speak to this netlify issue?

File: .netlify/build-and-move.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+bundle exec middleman build
```
Comment (DingoEatingFuzz) [inline .netlify/build-and-move.sh]: I'm sure this is fine, and I don't know what the equivalent would be if we were to use the docker container setup, but this difference between local development and netlify build gives me pause.

File: netlify.toml [hunk 3]
```diff
@@ -1,18 +1,6 @@
...
-# This ensures that following the link from the PR checks list redirects to the subdirectory.
-[[redirects]]
-  from = "/"
-  to = "/ui/"
-
-# This catches other unknown URLs and lets Ember handle them.
-[[redirects]]
-  from = "/*"
-  to = "/ui/index.html"
-  status = 200
```
Comment (DingoEatingFuzz) [inline netlify.toml]: It's nice that netlify also offers the `_redirects` file option.

**Comment by backspace** [comment]
Hey, @DingoEatingFuzz, I’m sorry to have requested review apparently too soon, as someone from Netlify let me know that it’s actually possible to have two different deployments for the same repository. I think that’ll be a better approach as each deployment can proceed from the relevant subdirectory, which will preserve caching. I attempted to retract the review request but I guess it didn’t take! So I’ll look into this option today, I have high hopes 🤞

**Comment by DingoEatingFuzz** [comment]
Oooooh, awesome!

**Comment by backspace** [comment]
Closing in favour of #6194.

**Pull request closed without merging**